tests/testsuite/bad_config.rs

```diff
@@ -983,6 +1071,42 @@ fn lib_crate_type2() {
         .run();
 }
 
+#[cargo_test(nightly, reason = "edition2024 is not stable")]
+fn lib_crate_type2_2024() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                cargo-features = ["edition2024"]
+
+                [package]
+                name = "foo"
+                version = "0.5.0"
+                edition = "2024"
+                authors = ["wycats@example.com"]
```

Comment:
How about we do this in a separate pass?  This would add a lot of churn to this PR for a superficial test change that is no worse than all of our other tests.